flake.nix

```diff
@@ -0,0 +1,61 @@
+{
+  description = "Dev environment + build for vine";
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { nixpkgs, rust-overlay, ... }: 
+  let
+    supportedSystems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+    pkgs = system: (import nixpkgs { 
+      inherit system; 
+      overlays = [ rust-overlay.overlays.default ]; 
+    }); 
+  in
+  {
+    devShells = forAllSystems (system: with (pkgs system);
+      {
+        default = mkShell {
+          buildInputs = [ rustup ];
+          shellHook = ''
+            source <(rustup completions bash)
+            source <(rustup completions bash cargo)
+          '';
+        };
+      }
```

Comment:
I think this is fine if one uses rustup, not necessarily the standard configuration though.

Reply:
Fair.  Is there a specific preference?  I used this because it was easy to setup without any other strong opinions.